### ISSUE TITLE
Duplicated errors with `nu --ide-check`

### ISSUE THREAD

**Issue by blindFS** [nushell/nushell#14697]
### Describe the bug

`nu --ide-check` and `nu --lsp` will generate duplicated copies of the same error, sometimes up-to 5 times (I don't remember how to reproduce 5x though)

It's not a big deal, but I guess there's some potential performance optimization opportunities here in the parser.

### How to reproduce

1. create a file named test.nu:
```nushell
use foo/bar *
$foo
def 
```

2. run `nu -n --ide-check 99 test.nu`

```json
{"type":"diagnostic","severity":"Error","message":"Module not found.","span":{"start":4,"end":11}}
{"type":"diagnostic","severity":"Error","message":"Module not found.","span":{"start":4,"end":11}}
{"type":"diagnostic","severity":"Error","message":"Variable not found.","span":{"start":14,"end":18}}
{"type":"diagnostic","severity":"Error","message":"Missing required positional argument.","span":{"start":22,"end":22}}
{"type":"diagnostic","severity":"Error","message":"Missing required positional argument.","span":{"start":22,"end":22}}
```

### Expected behavior

single entry for each error.

### Configuration

| key                | value                                                             |
| ------------------ | ----------------------------------------------------------------- |
| version            | 0.101.0                                                           |
| major              | 0                                                                 |
| minor              | 101                                                               |
| patch              | 0                                                                 |
| branch             |                                                                   |
| commit_hash        |                                                                   |
| build_os           | macos-aarch64                                                     |
| build_target       | aarch64-apple-darwin                                              |
| rust_version       | rustc 1.83.0 (90b35a623 2024-11-26) (built from a source tarball) |
| cargo_version      | cargo 1.83.0 (5ffbef321 2024-10-29)                               |
| build_time         | 1980-01-01 00:00:00 +00:00                                        |
| build_rust_channel | release                                                           |
| allocator          | mimalloc                                                          |
| features           | default, sqlite, trash                                            |
| installed_plugins  |                                                                   |

**Comment by fdncred** [comment]
That's weird. I could see having duplicate errors in the vscode extension code since validate is called nearly on every character you type. I remember putting in some debouncing to prevent that some though.

But if you just run `nu -n --ide-check 99 test.nu` on a file that isn't changing, it shouldn't happen. It's interesting that it does. I have no idea how that can happen.

**Comment by blindFS** [comment]
Take `Module not found` for example,

1st error
https://github.com/nushell/nushell/blob/2bcf2389aa1993c65b2353f6046a6398ce28128b/crates/nu-parser/src/parse_keywords.rs#L2400-L2400

2nd
https://github.com/nushell/nushell/blob/2bcf2389aa1993c65b2353f6046a6398ce28128b/crates/nu-parser/src/parse_keywords.rs#L2422

**Comment by blindFS** [comment]
@fdncred yes, but inside the function, an error is already emitted before returning none.

**Comment by fdncred** [comment]
oh, inside the parse_module_file_or_dir function there can be errors too. I see. What is your suggestion?

**Comment by blindFS** [comment]
I'm not sure, seems like a common issue elsewhere. Can be time consuming to identify them all. How about dedup before the final output?

**Comment by fdncred** [comment]
I'd rather have the dupes than just hide them. Maybe a tracking issue could be created, and we can update it as items are fixed?

**Comment by blindFS** [comment]
I'll try to find a way to get a comprehensive list of them later.

**Comment by blindFS** [comment]
I've found (through randomly perverted code from nu_scripts) following errors that can be duplicated somehow.

* [ ] Assignment operations require a variable.
* [ ] Assignment to an immutable variable.
* [ ] Capture of mutable variable.
* [ ] Command does not support ...
* [ ] Command name not support ...
* [ ] Default value is the wrong type
* [ ] Default values should be constant expressions.
* [ ] Error: nu::shell::column_defined_twice
* [ ] Error: nu::shell::not_a_constant
* [ ] Expected keyword.
* [ ] External command calls must be explicit in assignments
* [ ] Extra characters in the parameter name
* [ ] Extra columns.
* [ ] Extra positional argument.
* [ ] File not found
* [ ] Incomplete math expression.
* [ ] Invalid characters after closing delimiter
* [ ] Invalid literal
* [ ] Missing argument to ``.
* [ ] Missing columns.
* [ ] Missing required positional argument.
* [ ] Missing type.
* [ ] Module not found.
* [ ] Multiple rest params.
* [ ] Parse mismatch during operation.
* [ ] Parser incomplete.
* [ ] Required positional parameter after optional parameter
* [ ] Rest parameter needs a name.
* [ ] Statement used in pipeline.
* [ ] Table item not list
* [ ] The '' operator is not support ...
* [ ] The `` command doesn't have flag ``.
* [ ] This command does not have a ...rest parameter
* [ ] Type mismatch during operation.
* [ ] Type mismatch.
* [ ] Unbalanced delimiter.
* [ ] Unclosed delimiter.
* [ ] Unexpected custom completer in type spec
* [ ] Unexpected end of code.
* [ ] Unexpected keyword.
* [ ] Unknown command.
* [ ] Unknown type.
* [ ] Use $env.config instead of $config.
* [ ] Variable not found.
* [ ] boolean operation is not support ...
* [ ] filesize value must be a number
* [ ] let statement used in pipeline.
